controllers/servicefence_controller.go

```diff
@@ -114,8 +126,17 @@ func (r *ServicefenceReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 		if errors.IsNotFound(err) {
 			// TODO should be recovered? maybe we should call refreshFenceStatusOfService here
 			log.Info("serviceFence is deleted")
-			r.source.WatchRemove(req.NamespacedName)
-			return reconcile.Result{}, nil
+			//r.interestMeta.Pop(req.NamespacedName.String())
+			delete(r.interestMeta, req.NamespacedName.String())
+			for k, v := range r.interestMeta {
+				r.InterestMetaCopy[k] = v
```

Comment:
wrong usage

Reply:
changed according to concept of copy on write